src/components/swap/ConfirmSwapModal.tsx

```diff
@@ -39,22 +39,30 @@ export default function ConfirmSwapModal({
   swapErrorMessage: ReactNode | undefined
   onDismiss: () => void
 }) {
+  const [shouldLogModalCloseEvent, setShouldLogModalCloseEvent] = useState(false)
```

Comment:
i don't think i'm following the logic here

Reply:
just add a comment for what `shouldLogModalCloseEvent` means and why it needs to be passed into a child. i did end up figuring out what's going on.

Reply:
sgtm, added.